core/domain/suggestion_registry.py

```diff
@@ -753,27 +787,21 @@ def validate(self):
         question.partial_validate()
         question_state_data_schema_version = (
             self.change.question_dict['question_state_data_schema_version'])
-        if not (
-                question_state_data_schema_version >= 1 and
-                question_state_data_schema_version <=
+        if question_state_data_schema_version != (
                 feconf.CURRENT_STATE_SCHEMA_VERSION):
             raise utils.ValidationError(
-                'Expected question state schema version to be between 1 and '
-                '%s' % feconf.CURRENT_STATE_SCHEMA_VERSION)
+                'Expected question state schema version to be %s, received '
+                '%s' % (
+                    feconf.CURRENT_STATE_SCHEMA_VERSION,
+                    question_state_data_schema_version))
 
     def pre_accept_validate(self):
         """Performs referential validation. This function needs to be called
         before accepting the suggestion.
         """
         if self.change.skill_id is None:
             raise utils.ValidationError('Expected change to contain skill_id')
-        question_dict = self.change.question_dict
         self.validate()
-        if (
-                question_dict['question_state_data_schema_version'] !=
```

Comment:
For reviewers:

Removed this check as we have added this check inside the validate method*